nav2_smac_planner/include/nav2_smac_planner/smac_planner_hybrid.hpp

```diff
@@ -87,21 +87,41 @@ class SmacPlannerHybrid : public nav2_core::GlobalPlanner
     const geometry_msgs::msg::PoseStamped & goal) override;
 
 protected:
+  /**
+   * @brief Callback executed when a parameter change is detected
+   * @param event ParameterEvent message
+   */
+  void on_parameter_event_callback(const rcl_interfaces::msg::ParameterEvent::SharedPtr event);
+
   std::unique_ptr<AStarAlgorithm<NodeHybrid>> _a_star;
   GridCollisionChecker _collision_checker;
   std::unique_ptr<Smoother> _smoother;
   rclcpp::Clock::SharedPtr _clock;
   rclcpp::Logger _logger{rclcpp::get_logger("SmacPlannerHybrid")};
   nav2_costmap_2d::Costmap2D * _costmap;
+  std::shared_ptr<nav2_costmap_2d::Costmap2DROS> _costmap_ros;
   std::unique_ptr<CostmapDownsampler> _costmap_downsampler;
   std::string _global_frame, _name;
+  float _lookup_table_dim;
   float _tolerance;
+  bool _downsample_costmap;
   int _downsampling_factor;
-  unsigned int _angle_quantizations;
   double _angle_bin_size;
-  bool _downsample_costmap;
-  rclcpp_lifecycle::LifecyclePublisher<nav_msgs::msg::Path>::SharedPtr _raw_plan_publisher;
+  int _angle_quantizations;
```

Comment:
You changed the type of this, it should be unsigned

Reply:
But then I cast to `unsigned int` as it was done before

Reply:
I'd just have this keep the types consistent, there are really easy errors within this library that can be introduced if the signed / unsigned ness are messed up, so I'm pretty careful about it.

Reply:
It can be an int from the parameter, but see how it was being used before, it gets  cast immediately and used as unsigned from there on out

Reply:
You had deleted that logic for some reason

Reply:
Got it. Fixed in last commit.